File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.cpp
class Solution {
  public:
    int maxSubarraySum(vector<int> &arr) {
        // Code here
        int n=arr.size();
        int currsum=0;
        int ans=INT_MIN;
        for(int num:arr){
            currsum+=num;
            ans=max(ans,currsum);
            if(currsum<0)
            currsum=0;
        }
        return ans;
    }
};

File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.java
class Solution {
    int maxSubarraySum(int[] arr) {
        // Code here
        int maxsum=arr[0];
        int currsum=arr[0];
        for(int i=1;i<arr.length;i++){
            currsum=Math.max(arr[i],currsum+arr[i]);
            maxsum=Math.max(maxsum,currsum);
        }
        return maxsum;
    }
}
